open-api/lang-parser/lang-parser.gradle

```diff
@@ -12,6 +12,8 @@ apply from: "$rootDir/gradle/web-common.gradle"
 dependencies {
     compile project(':open-api-shared-ide')
     compile project(':open-api-shared-fu')
+    compile project(':IO')
+    compile project(':proto:proto-backplane-grpc')
```

Comment:
not an "issue", just a remark:

in most other parts of the repo, we've tried to keep proto contained to grpc-api, so that the semantics of how you make calls doesn't creep into other aspects of the codebase - provided nothing else (like db) depends on lang-parser I don't think this dep is a problem, but just pointing out the decision that was made.

ditto dagger2, we're keeping that in grpc-api, and just encouraging other code to use patterns that make it easy for code to have dependencies injected, but not require consumers of the codebase to use our di

Reply:
ah whoops, db does_ now depend on open-api-lang-parser... may need to revisit this soon if not right away, make sure this is unavoidable

Reply:
I had a diamond problem between the parser and the completer.  The parser operates on "lsp types", so I used the grpc generated types.  The completer needs objects like Table.  The completer also needs the parser.  

Advice on what to do here?  If / when Table and TableDefinition (and VariableProvider) can be moved out of db, then I could resolve this, but, as is, unless I want to have three copies of the same objects in slightly different formats spread out across all the modules, I dunno how else to fix.

Reply:
Well, the completer mostly operates on the lsp types, but I put utilities into the AST nodes to make them able to understand / translate between AST and LSP.  I _might_ be able to extract them all into static util methods; annoything, but probably not impossible. I'll take a quick spin at it.

Reply:
oh man, there's lots and lots and lots of references :-/

Reply:
I think we should merge it like this, and just open a ticket to refactor these dependencies once the Table interface is split up, and things are ....less likely to suck in the whole world.

Reply:
Ok, pushed commit w/ a refactor.  It's a little ugly w/ the type parameters, but the wider, shorter dependency tree is nicer.